File: src/game/cards.ts
export type CardId =
  | "knight"
  | "archers"
  | "giant"
  | "musketeer"
  | "mini-pekka"
  | "skeletons"
  | "fireball"
  | "arrows"
  | "zap"
  | "rage"
  | "freeze"
  | "wizard"
  | "witch"
  | "hog-rider"
  | "balloon"
  | "baby-dragon"
  | "gargoyles"
  | "valkyrie"
  | "prince"
  | "pekka"
  | "cannon"
  | "tombstone"
  | "elixir-collector";

export type Speed = "slow" | "medium" | "fast";

/** CR-style card rarity, shown as the card frame color. */
export type Rarity = "common" | "rare" | "epic";

/** Tiles per second for each named speed. */
export const SPEED_TILES_PER_SEC: Record<Speed, number> = {
  slow: 0.75,
  medium: 1.1,
  fast: 1.6,
};

export interface UnitStats {
  maxHp: number;
  damage: number;
  /** Seconds between attacks. */
  hitSpeed: number;
  /** Distance (tiles) at which the unit can attack. Melee ≈ 0.8. */
  attackRange: number;
  /** Distance (tiles) at which the unit notices enemy troops. */
  sightRange: number;
  speed: Speed;
  targetsBuildingsOnly: boolean;
  /** Can this unit hit flying targets? */
  targetsAir: boolean;
  /** Does this unit fly (straight paths, only hit by air-targeters)? */
  flying: boolean;
  /** Ground unit that leaps the river instead of detouring to a bridge. */
  jumpsRiver: boolean;
  /** Area damage around the struck target; 0 = single target. */
  splashRadius: number;
  /** Tiles of uninterrupted approach to charge (0 = no charge). */
  chargeDistance: number;
  /** Damage dealt to nearby enemies when this unit dies (0 = none). */
  deathDamage: number;
  /** Radius of the death blast in tiles. */
  deathRadius: number;
  /** Card whose units this troop periodically summons (null = none). */
  spawnUnitId: CardId | null;
  /** Seconds between summons (0 = not a spawner). */
  spawnInterval: number;
  /** Seconds per 1 elixir generated for the owner (0 = none). */
  elixirInterval: number;
  /** Visual + collision radius in tiles. */
  radius: number;
}

export interface TroopCard {
  id: CardId;
  name: string;
  rarity: Rarity;
  kind: "troop";
  cost: number;
  /** Number of units spawned per deploy. */
  count: number;
  unit: UnitStats;
}

export interface BuildingCard {
  id: CardId;
  name: string;
  rarity: Rarity;
  kind: "building";
  cost: number;
  /** Seconds before the building expires on its own. */
  lifetime: number;
  unit: UnitStats;
}

export interface SpellCard {
  id: CardId;
  name: string;
  rarity: Rarity;
  kind: "spell";
  cost: number;
  damage: number;
  radius: number;
  /** Seconds enemies hit are frozen in place (0 = no stun). */
  stunSeconds: number;
  /** Seconds the spell leaves a friendly speed-boost zone (0 = none). */
  rageSeconds: number;
  /** Tiles survivors are shoved away from the blast center. */
  knockback: number;
}

export type Card = TroopCard | BuildingCard | SpellCard;

const MELEE = 0.8;

interface UnitOverrides extends Partial<UnitStats> {
  maxHp: number;
  damage: number;
  hitSpeed: number;
  attackRange: number;
  speed: Speed;
}

function unit(stats: UnitOverrides): UnitStats {
  return {
    sightRange: 5.5,
    targetsBuildingsOnly: false,
    targetsAir: false,
    flying: false,
    jumpsRiver: false,
    splashRadius: 0,
    chargeDistance: 0,
    deathDamage: 0,
    deathRadius: 0,
    spawnUnitId: null,
    spawnInterval: 0,
    elixirInterval: 0,
    radius: 0.5,
    ...stats,
  };
}

export const CARDS: Record<CardId, Card> = {
  knight: {
    id: "knight",
    name: "Knight",
    rarity: "common",
    kind: "troop",
    cost: 3,
    count: 1,
    unit: unit({
      maxHp: 1400,
      damage: 160,
      hitSpeed: 1.2,
      attackRange: MELEE,
      speed: "medium",
    }),
  },
  archers: {
    id: "archers",
    name: "Archers",
    rarity: "common",
    kind: "troop",
    cost: 3,
    count: 2,
    unit: unit({
      maxHp: 250,
      damage: 90,
      hitSpeed: 1.2,
      attackRange: 5,
      speed: "medium",
      targetsAir: true,
      radius: 0.4,
    }),
  },
  giant: {
    id: "giant",
    name: "Giant",
    rarity: "rare",
    kind: "troop",
    cost: 5,
    count: 1,
    unit: unit({
      maxHp: 3300,
      damage: 210,
      hitSpeed: 1.5,
      attackRange: MELEE,
      sightRange: 7.5,
      speed: "slow",
      targetsBuildingsOnly: true,
      radius: 0.75,
    }),
  },
  musketeer: {
    id: "musketeer",
    name: "Musketeer",
    rarity: "rare",
    kind: "troop",
    cost: 4,
    count: 1,
    unit: unit({
      maxHp: 600,
      damage: 180,
      hitSpeed: 1.1,
      attackRange: 6,
      sightRange: 6,
      speed: "medium",
      targetsAir: true,
    }),
  },
  "mini-pekka": {
    id: "mini-pekka",
    name: "Mini P.E.K.K.A",
    rarity: "rare",
    kind: "troop",
    cost: 4,
    count: 1,
    unit: unit({
      maxHp: 1100,
      damage: 600,
      hitSpeed: 1.8,
      attackRange: MELEE,
      speed: "fast",
    }),
  },
  skeletons: {
    id: "skeletons",
    name: "Skeletons",
    rarity: "common",
    kind: "troop",
    cost: 1,
    count: 3,
    unit: unit({
      maxHp: 80,
      damage: 80,
      hitSpeed: 1.0,
      attackRange: MELEE,
      speed: "fast",
      radius: 0.3,
    }),
  },
  wizard: {
    id: "wizard",
    name: "Wizard",
    rarity: "rare",
    kind: "troop",
    cost: 5,
    count: 1,
    unit: unit({
      maxHp: 600,
      damage: 230,
      hitSpeed: 1.4,
      attackRange: 5.5,
      sightRange: 6,
      speed: "medium",
      targetsAir: true,
      splashRadius: 1.2,
    }),
  },
  witch: {
    id: "witch",
    name: "Witch",
    rarity: "epic",
    kind: "troop",
    cost: 5,
    count: 1,
    unit: unit({
      maxHp: 700,
      damage: 130,
      hitSpeed: 1.1,
      attackRange: 5,
      sightRange: 5.5,
      speed: "medium",
      targetsAir: true,
      splashRadius: 1.0,
      spawnUnitId: "skeletons",
      spawnInterval: 7,
    }),
  },
  "hog-rider": {
    id: "hog-rider",
    name: "Hog Rider",
    rarity: "rare",
    kind: "troop",
    cost: 4,
    count: 1,
    unit: unit({
      maxHp: 1500,
      damage: 260,
      hitSpeed: 1.6,
      attackRange: MELEE,
      sightRange: 7.5,
      speed: "fast",
      targetsBuildingsOnly: true,
      jumpsRiver: true,
      radius: 0.6,
    }),
  },
  balloon: {
    id: "balloon",
    name: "Balloon",
    rarity: "epic",
    kind: "troop",
    cost: 5,
    count: 1,
    unit: unit({
      maxHp: 1500,
      damage: 600,
      hitSpeed: 3,
      attackRange: MELEE,
      sightRange: 7.5,
      speed: "medium",
      targetsBuildingsOnly: true,
      flying: true,
      deathDamage: 300,
      deathRadius: 1.5,
      radius: 0.7,
    }),
  },
  "baby-dragon": {
    id: "baby-dragon",
    name: "Baby Dragon",
    rarity: "epic",
    kind: "troop",
    cost: 4,
    count: 1,
    unit: unit({
      maxHp: 1050,
      damage: 130,
      hitSpeed: 1.5,
      attackRange: 3.5,
      speed: "fast",
      targetsAir: true,
      flying: true,
      splashRadius: 1.0,
      radius: 0.6,
    }),
  },
  gargoyles: {
    id: "gargoyles",
    name: "Gargoyles",
    rarity: "common",
    kind: "troop",
    cost: 3,
    count: 3,
    unit: unit({
      maxHp: 190,
      damage: 85,
      hitSpeed: 1.0,
      attackRange: MELEE,
      speed: "fast",
      targetsAir: true,
      flying: true,
      radius: 0.35,
    }),
  },
  valkyrie: {
    id: "valkyrie",
    name: "Valkyrie",
    rarity: "rare",
    kind: "troop",
    cost: 4,
    count: 1,
    unit: unit({
      maxHp: 1500,
      damage: 220,
      hitSpeed: 1.5,
      attackRange: MELEE,
      speed: "medium",
      splashRadius: 1.2,
    }),
  },
  prince: {
    id: "prince",
    name: "Prince",
    rarity: "epic",
    kind: "troop",
    cost: 5,
    count: 1,
    unit: unit({
      maxHp: 1500,
      damage: 325,
      hitSpeed: 1.4,
      attackRange: MELEE,
      speed: "medium",
      chargeDistance: 2.5,
      radius: 0.6,
    }),
  },
  pekka: {
    id: "pekka",
    name: "P.E.K.K.A",
    rarity: "epic",
    kind: "troop",
    cost: 7,
    count: 1,
    unit: unit({
      maxHp: 3000,
      damage: 750,
      hitSpeed: 1.8,
      attackRange: MELEE,
      speed: "slow",
      radius: 0.7,
    }),
  },
  cannon: {
    id: "cannon",
    name: "Cannon",
    rarity: "common",
    kind: "building",
    cost: 3,
    lifetime: 30,
    unit: unit({
      maxHp: 800,
      damage: 130,
      hitSpeed: 0.9,
      attackRange: 5.5,
      speed: "slow", // unused: buildings don't move
      radius: 0.6,
    }),
  },
  tombstone: {
    id: "tombstone",
    name: "Tombstone",
    rarity: "rare",
    kind: "building",
    cost: 3,
    lifetime: 30,
    unit: unit({
      maxHp: 600,
      damage: 0, // it just spawns, never attacks
      hitSpeed: 1,
      attackRange: 0,
      speed: "slow", // unused: buildings don't move
      spawnUnitId: "skeletons",
      spawnInterval: 6,
      radius: 0.6,
    }),
  },
  "elixir-collector": {
    id: "elixir-collector",
    name: "Elixir Collector",
    rarity: "rare",
    kind: "building",
    cost: 6,
    lifetime: 70,
    unit: unit({
      maxHp: 900,
      damage: 0, // pure economy, never attacks
      hitSpeed: 1,
      attackRange: 0,
      speed: "slow", // unused: buildings don't move
      elixirInterval: 8.5,
      radius: 0.6,
    }),
  },
  fireball: {
    id: "fireball",
    name: "Fireball",
    rarity: "rare",
    kind: "spell",
    cost: 4,
    damage: 570,
    radius: 2.5,
    stunSeconds: 0,
    rageSeconds: 0,
    knockback: 0.8,
  },
  arrows: {
    id: "arrows",
    name: "Arrows",
    rarity: "common",
    kind: "spell",
    cost: 3,
    damage: 240,
    radius: 4,
    stunSeconds: 0,
    rageSeconds: 0,
    knockback: 0,
  },
  zap: {
    id: "zap",
    name: "Zap",
    rarity: "common",
    kind: "spell",
    cost: 2,
    damage: 250,
    radius: 2,
    stunSeconds: 0.5,
    rageSeconds: 0,
    knockback: 0,
  },
  rage: {
    id: "rage",
    name: "Rage",
    rarity: "epic",
    kind: "spell",
    cost: 2,
    damage: 0,
    radius: 2.5,
    stunSeconds: 0,
    rageSeconds: 6,
    knockback: 0,
  },
  freeze: {
    id: "freeze",
    name: "Freeze",
    rarity: "epic",
    kind: "spell",
    cost: 4,
    damage: 0,
    radius: 3,
    stunSeconds: 4,
    rageSeconds: 0,
    knockback: 0,
  },
};

/** Deck order doubles as the starting draw: the first 4 are the opening hand. */
export const DECK: CardId[] = [
  "knight",
  "archers",
  "giant",
  "fireball",
  "musketeer",
  "mini-pekka",
  "baby-dragon",
  "valkyrie",
  "skeletons",
  "wizard",
  "witch",
  "hog-rider",
  "balloon",
  "prince",
  "pekka",
  "cannon",
  "tombstone",
  "elixir-collector",
  "gargoyles",
  "arrows",
  "zap",
  "rage",
  "freeze",
];

export function getCard(id: CardId): Card {
  return CARDS[id];
}

/**
 * CR-style battle deck: exactly 8 cards. The first 4 are the opening
 * hand; this starter deck keeps the classic opening (knight, archers,
 * giant, fireball).
 */
export const DEFAULT_DECK: CardId[] = [
  "knight",
  "archers",
  "giant",
  "fireball",
  "musketeer",
  "mini-pekka",
  "baby-dragon",
  "arrows",
];

File: unity/ClashRoyaleUnity/Assets/Scripts/Sim/Cards.cs
using System.Collections.Generic;

namespace ClashRoyale.Sim
{
    public enum CardId
    {
        Knight,
        Archers,
        Giant,
        Musketeer,
        MiniPekka,
        Skeletons,
        Fireball,
        Arrows,
        Zap,
        Rage,
        Freeze,
        Wizard,
        Witch,
        HogRider,
        Balloon,
        BabyDragon,
        Gargoyles,
        Valkyrie,
        Prince,
        Pekka,
        Cannon,
        Tombstone,
        ElixirCollector,
    }

    public enum Speed
    {
        Slow,
        Medium,
        Fast,
    }

    /// <summary>CR-style card rarity, shown as the card frame color.</summary>
    public enum Rarity
    {
        Common,
        Rare,
        Epic,
    }

    public enum CardKind
    {
        Troop,
        Building,
        Spell,
    }

    /// <summary>Per-unit combat stats, ported from <c>UnitStats</c> in cards.ts.</summary>
    public sealed class UnitStats
    {
        public double MaxHp;
        public double Damage;

        /// <summary>Seconds between attacks.</summary>
        public double HitSpeed;

        /// <summary>Distance (tiles) at which the unit can attack. Melee ~= 0.8.</summary>
        public double AttackRange;

        /// <summary>Distance (tiles) at which the unit notices enemy troops.</summary>
        public double SightRange = 5.5;

        public Speed Speed;
        public bool TargetsBuildingsOnly;
        public bool TargetsAir;
        public bool Flying;
        public bool JumpsRiver;
        public double SplashRadius;
        public double ChargeDistance;
        public double DeathDamage;
        public double DeathRadius;
        public CardId? SpawnUnitId;
        public double SpawnInterval;
        public double ElixirInterval;
        public double Radius = 0.5;
    }

    public abstract class Card
    {
        public CardId Id;
        public string Name;
        public Rarity Rarity;
        public CardKind Kind;
        public int Cost;
    }

    public sealed class TroopCard : Card
    {
        /// <summary>Number of units spawned per deploy.</summary>
        public int Count;
        public UnitStats Unit;
    }

    public sealed class BuildingCard : Card
    {
        /// <summary>Seconds before the building expires on its own.</summary>
        public double Lifetime;
        public UnitStats Unit;
    }

    public sealed class SpellCard : Card
    {
        public double Damage;
        public double Radius;
        public double StunSeconds;
        public double RageSeconds;
        public double Knockback;
    }

    public static class Cards
    {
        public const double Melee = 0.8;

        /// <summary>Tiles per second for each named speed.</summary>
        public static readonly Dictionary<Speed, double> SpeedTilesPerSec = new()
        {
            { Speed.Slow, 0.75 },
            { Speed.Medium, 1.1 },
            { Speed.Fast, 1.6 },
        };

        public static readonly Dictionary<CardId, Card> All = BuildCards();

        public static Card Get(CardId id)
        {
            return All[id];
        }

        /// <summary>
        /// Deck order doubles as the starting draw: the first 4 are the opening hand.
        /// </summary>
        public static readonly List<CardId> Deck = new()
        {
            CardId.Knight, CardId.Archers, CardId.Giant, CardId.Fireball,
            CardId.Musketeer, CardId.MiniPekka, CardId.BabyDragon, CardId.Valkyrie,
            CardId.Skeletons, CardId.Wizard, CardId.Witch, CardId.HogRider,
            CardId.Balloon, CardId.Prince, CardId.Pekka, CardId.Cannon,
            CardId.Tombstone, CardId.ElixirCollector, CardId.Gargoyles, CardId.Arrows,
            CardId.Zap, CardId.Rage, CardId.Freeze,
        };

        /// <summary>CR-style 8-card battle deck; first 4 are the opening hand.</summary>
        public static readonly List<CardId> DefaultDeck = new()
        {
            CardId.Knight, CardId.Archers, CardId.Giant, CardId.Fireball,
            CardId.Musketeer, CardId.MiniPekka, CardId.BabyDragon, CardId.Arrows,
        };

        private static TroopCard Troop(CardId id, string name, Rarity rarity, int cost, int count, UnitStats unit)
        {
            return new TroopCard
            {
                Id = id, Name = name, Rarity = rarity, Kind = CardKind.Troop,
                Cost = cost, Count = count, Unit = unit,
            };
        }

        private static BuildingCard Building(CardId id, string name, Rarity rarity, int cost, double lifetime, UnitStats unit)
        {
            return new BuildingCard
            {
                Id = id, Name = name, Rarity = rarity, Kind = CardKind.Building,
                Cost = cost, Lifetime = lifetime, Unit = unit,
            };
        }

        private static SpellCard Spell(CardId id, string name, Rarity rarity, int cost,
            double damage, double radius, double stunSeconds, double rageSeconds, double knockback)
        {
            return new SpellCard
            {
                Id = id, Name = name, Rarity = rarity, Kind = CardKind.Spell, Cost = cost,
                Damage = damage, Radius = radius, StunSeconds = stunSeconds,
                RageSeconds = rageSeconds, Knockback = knockback,
            };
        }

        private static Dictionary<CardId, Card> BuildCards()
        {
            return new Dictionary<CardId, Card>
            {
                [CardId.Knight] = Troop(CardId.Knight, "Knight", Rarity.Common, 3, 1, new UnitStats
                {
                    MaxHp = 1400, Damage = 160, HitSpeed = 1.2, AttackRange = Melee, Speed = Speed.Medium,
                }),
                [CardId.Archers] = Troop(CardId.Archers, "Archers", Rarity.Common, 3, 2, new UnitStats
                {
                    MaxHp = 250, Damage = 90, HitSpeed = 1.2, AttackRange = 5, Speed = Speed.Medium,
                    TargetsAir = true, Radius = 0.4,
                }),
                [CardId.Giant] = Troop(CardId.Giant, "Giant", Rarity.Rare, 5, 1, new UnitStats
                {
                    MaxHp = 3300, Damage = 210, HitSpeed = 1.5, AttackRange = Melee, SightRange = 7.5,
                    Speed = Speed.Slow, TargetsBuildingsOnly = true, Radius = 0.75,
                }),
                [CardId.Musketeer] = Troop(CardId.Musketeer, "Musketeer", Rarity.Rare, 4, 1, new UnitStats
                {
                    MaxHp = 600, Damage = 180, HitSpeed = 1.1, AttackRange = 6, SightRange = 6,
                    Speed = Speed.Medium, TargetsAir = true,
                }),
                [CardId.MiniPekka] = Troop(CardId.MiniPekka, "Mini P.E.K.K.A", Rarity.Rare, 4, 1, new UnitStats
                {
                    MaxHp = 1100, Damage = 600, HitSpeed = 1.8, AttackRange = Melee, Speed = Speed.Fast,
                }),
                [CardId.Skeletons] = Troop(CardId.Skeletons, "Skeletons", Rarity.Common, 1, 3, new UnitStats
                {
                    MaxHp = 80, Damage = 80, HitSpeed = 1.0, AttackRange = Melee, Speed = Speed.Fast, Radius = 0.3,
                }),
                [CardId.Wizard] = Troop(CardId.Wizard, "Wizard", Rarity.Rare, 5, 1, new UnitStats
                {
                    MaxHp = 600, Damage = 230, HitSpeed = 1.4, AttackRange = 5.5, SightRange = 6,
                    Speed = Speed.Medium, TargetsAir = true, SplashRadius = 1.2,
                }),
                [CardId.Witch] = Troop(CardId.Witch, "Witch", Rarity.Epic, 5, 1, new UnitStats
                {
                    MaxHp = 700, Damage = 130, HitSpeed = 1.1, AttackRange = 5, SightRange = 5.5,
                    Speed = Speed.Medium, TargetsAir = true, SplashRadius = 1.0,
                    SpawnUnitId = CardId.Skeletons, SpawnInterval = 7,
                }),
                [CardId.HogRider] = Troop(CardId.HogRider, "Hog Rider", Rarity.Rare, 4, 1, new UnitStats
                {
                    MaxHp = 1500, Damage = 260, HitSpeed = 1.6, AttackRange = Melee, SightRange = 7.5,
                    Speed = Speed.Fast, TargetsBuildingsOnly = true, JumpsRiver = true, Radius = 0.6,
                }),
                [CardId.Balloon] = Troop(CardId.Balloon, "Balloon", Rarity.Epic, 5, 1, new UnitStats
                {
                    MaxHp = 1500, Damage = 600, HitSpeed = 3, AttackRange = Melee, SightRange = 7.5,
                    Speed = Speed.Medium, TargetsBuildingsOnly = true, Flying = true,
                    DeathDamage = 300, DeathRadius = 1.5, Radius = 0.7,
                }),
                [CardId.BabyDragon] = Troop(CardId.BabyDragon, "Baby Dragon", Rarity.Epic, 4, 1, new UnitStats
                {
                    MaxHp = 1050, Damage = 130, HitSpeed = 1.5, AttackRange = 3.5, Speed = Speed.Fast,
                    TargetsAir = true, Flying = true, SplashRadius = 1.0, Radius = 0.6,
                }),
                [CardId.Gargoyles] = Troop(CardId.Gargoyles, "Gargoyles", Rarity.Common, 3, 3, new UnitStats
                {
                    MaxHp = 190, Damage = 85, HitSpeed = 1.0, AttackRange = Melee, Speed = Speed.Fast,
                    TargetsAir = true, Flying = true, Radius = 0.35,
                }),
                [CardId.Valkyrie] = Troop(CardId.Valkyrie, "Valkyrie", Rarity.Rare, 4, 1, new UnitStats
                {
                    MaxHp = 1500, Damage = 220, HitSpeed = 1.5, AttackRange = Melee, Speed = Speed.Medium,
                    SplashRadius = 1.2,
                }),
                [CardId.Prince] = Troop(CardId.Prince, "Prince", Rarity.Epic, 5, 1, new UnitStats
                {
                    MaxHp = 1500, Damage = 325, HitSpeed = 1.4, AttackRange = Melee, Speed = Speed.Medium,
                    ChargeDistance = 2.5, Radius = 0.6,
                }),
                [CardId.Pekka] = Troop(CardId.Pekka, "P.E.K.K.A", Rarity.Epic, 7, 1, new UnitStats
                {
                    MaxHp = 2900, Damage = 650, HitSpeed = 1.8, AttackRange = Melee, Speed = Speed.Slow, Radius = 0.7,
                }),
                [CardId.Cannon] = Building(CardId.Cannon, "Cannon", Rarity.Common, 3, 30, new UnitStats
                {
                    MaxHp = 800, Damage = 130, HitSpeed = 0.9, AttackRange = 5.5, Speed = Speed.Slow, Radius = 0.6,
                }),
                [CardId.Tombstone] = Building(CardId.Tombstone, "Tombstone", Rarity.Rare, 3, 30, new UnitStats
                {
                    MaxHp = 600, Damage = 0, HitSpeed = 1, AttackRange = 0, Speed = Speed.Slow,
                    SpawnUnitId = CardId.Skeletons, SpawnInterval = 6, Radius = 0.6,
                }),
                [CardId.ElixirCollector] = Building(CardId.ElixirCollector, "Elixir Collector", Rarity.Rare, 6, 70, new UnitStats
                {
                    MaxHp = 900, Damage = 0, HitSpeed = 1, AttackRange = 0, Speed = Speed.Slow,
                    ElixirInterval = 8.5, Radius = 0.6,
                }),
                [CardId.Fireball] = Spell(CardId.Fireball, "Fireball", Rarity.Rare, 4, 570, 2.5, 0, 0, 0.8),
                [CardId.Arrows] = Spell(CardId.Arrows, "Arrows", Rarity.Common, 3, 240, 4, 0, 0, 0),
                [CardId.Zap] = Spell(CardId.Zap, "Zap", Rarity.Common, 2, 250, 2, 0.5, 0, 0),
                [CardId.Rage] = Spell(CardId.Rage, "Rage", Rarity.Epic, 2, 0, 2.5, 0, 6, 0),
                [CardId.Freeze] = Spell(CardId.Freeze, "Freeze", Rarity.Epic, 4, 0, 3, 4, 0, 0),
            };
        }
    }
}
